python-package/lightgbm/basic.py

```diff
@@ -1398,6 +1402,8 @@ def get_data(self):
                 self.data = self.data[self.used_indices, :]
             elif isinstance(self.data, DataFrame):
                 self.data = self.data.iloc[self.used_indices].copy()
+            elif isinstance(self.data, DataTable):
+                self.data = self.data[self.used_indices, :]
```

Comment:
Me too... To be honest, I see DataTable for the first time 😄 

Speaking about pandas, I used `.copy()` with the aim to prevent [`SettingWithCopyWarning`](https://www.dataquest.io/blog/settingwithcopywarning/) in users' preprocess functions.

Reply:
Okay, I see. So theoretically, we don't need a copy here.
@pseudotensor any idea about this?

Reply:
ping @pseudotensor 
Please take a look at this function for getting raw data.